contracts/contracts/strategies/MorphoAaveStrategy.sol

```diff
@@ -0,0 +1,239 @@
+
+/**
+ * @title OUSD Morpho Compound Strategy
+ * @notice Investment strategy for investing stablecoins via Morpho (Compound)
+ * @author Origin Protocol Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IERC20, InitializableAbstractStrategy } from "../utils/InitializableAbstractStrategy.sol";
+import { IMorpho } from "../interfaces/morpho/IMorpho.sol";
+import { ILens } from "../interfaces/morpho/ILens.sol";
+import { StableMath } from "../utils/StableMath.sol";
+import "../utils/Helpers.sol";
+
+contract MorphoAaveStrategy is InitializableAbstractStrategy {
+    address public constant MORPHO = 0x777777c9898D384F785Ee44Acfe945efDFf5f3E0;
+    address public constant LENS = 0x507fA343d0A90786d86C7cd885f5C49263A91FF4;
+
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
+    /**
+     * @dev Initialize function, to set up initial internal state
+     * @param _vaultAddress Address of the Vault
+     * @param _rewardTokenAddresses Address of reward token for platform
+     * @param _assets Addresses of initial supported assets
+     * @param _pTokens Platform Token corresponding addresses
+     */
+    function initialize(
+        address _vaultAddress,
+        address[] calldata _rewardTokenAddresses,
+        address[] calldata _assets,
+        address[] calldata _pTokens
+    ) external onlyGovernor initializer {
+        super._initialize(
+            MORPHO,
+            _vaultAddress,
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+    }
+
+    /**
+     * @dev Approve the spending of all assets by main Morpho contract,
+     *      if for some reason is it necessary.
+     */
+    function safeApproveAllTokens()
+        external
+        override
+        onlyGovernor
+        nonReentrant
+    {
+        uint256 assetCount = assetsMapped.length;
+        for (uint256 i = 0; i < assetCount; i++) {
+            address asset = assetsMapped[i];
+
+            // Safe approval
+            IERC20(asset).safeApprove(MORPHO, 0);
+            IERC20(asset).safeApprove(MORPHO, type(uint256).max);
+        }
+    }
+
+    /**
+     * @dev Internal method to respond to the addition of new asset
+     *      We need to approve and allow Morpho to move them
+     * @param _asset Address of the asset to approve
+     * @param _pToken The pToken for the approval
+     */
+    // solhint-disable-next-line no-unused-vars
+    function _abstractSetPToken(address _asset, address _pToken)
+        internal
+        override
+    {
+        IERC20(_asset).safeApprove(MORPHO, 0);
+        IERC20(_asset).safeApprove(MORPHO, type(uint256).max);
+    }
+
+    /**
+     * @dev Collect accumulated rewards and send them to Harvester.
+     */
+    function collectRewardTokens()
+        external
+        override
+        onlyHarvester
+        nonReentrant
+    {
+        // Morpho Aave-v2 doesn't distribute reward tokens
```

Comment:
Interestingly enough, AAVE reward collection and distrobution is in the current Morpho AAVE contract, but it's not configure to be active, and contract addresses needed are not set.

Because of incentivized harvesting, these two methods have no gas cost unless called, which would mean that we could go ahead and implement them.

However, since we can't test them currently, I'd much rather not have untested code that is live. I agree with you making these functions basically empty.

Reply:
> However, since we can't test them currently, I'd much rather not have untested code that is live

Agreed. We can always upgrade the contract when Morpho enables distributing rewards

Reply:
Currently, there are no rewards on Aave (and it's not planned). We'll remove them in a future upgrade

Reply:
thanks for info @MerlinEgalite